File: AFMobile/AFAndroid/app/src/main/java/cz/cvut/fel/matyapav/afandroid/components/parts/validators/AFValidator.java
package cz.cvut.fel.matyapav.afandroid.components.parts.validators;

import android.content.Context;

import cz.cvut.fel.matyapav.afandroid.components.parts.AFField;
import cz.cvut.fel.matyapav.afandroid.components.parts.ValidationRule;

/**
 * Validator interface
 *
 * @author Pavel Matyáš (matyapav@fel.cvut.cz).
 * @since 1.0.0..
 */
public interface AFValidator {

    boolean validate(Context context, AFField field, StringBuilder errorMsgs, ValidationRule rule);
}

File: AFMobile/AFWindowsPhone/AFWindowsPhone/components/parts/validators/AFValidator.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace AFWindowsPhone.builders.components.parts.validators
{
    interface AFValidator
    {
        bool validate(AFField field, StringBuilder errorMsgs, ValidationRule rule);
    }
}
